pkg/extensions/tbtc/tbtc.go

```diff
@@ -645,6 +744,84 @@ func (t *tbtc) shouldMonitorDeposit(depositAddress string) bool {
 	return false
 }
 
+func (t *tbtc) waitDepositStateChangeConfirmation(
+	depositAddress string,
+	initialDepositState chain.DepositState,
+) bool {
+	stateCheck := func() (bool, error) {
+		currentState, err := t.chain.CurrentState(depositAddress)
+		if err != nil {
+			return false, err
+		}
+
+		return currentState != initialDepositState, nil
+	}
+
+	currentBlock, err := t.chain.BlockCounter().CurrentBlock()
+	if err != nil {
+		logger.Errorf(
+			"could not get current block while confirming "+
+				"state [%v] change for deposit [%v]: [%v]",
+			initialDepositState,
+			depositAddress,
+			err,
+		)
+		return false
+	}
+
+	confirmed, err := ethutil.WaitForChainConfirmation(
+		t.chain.BlockCounter(),
+		currentBlock,
+		t.blockConfirmations,
+		stateCheck,
+	)
+	if err != nil {
+		logger.Errorf(
+			"could not confirm state [%v] change for deposit [%v]: [%v]",
+			initialDepositState,
+			depositAddress,
+			err,
+		)
+		return false
+	}
+
+	return confirmed
+}
+
+func (t *tbtc) waitKeepInactivityConfirmation(
```

Comment:
Can we rename it to `waitKeepNotActiveConfirmation`? *Inactivity* could suggest keep has not provided the requested signature and it's been terminated. *NotActive* relates more to the keep state.
(Still not perfect but I do not have any better idea).